src/objects/zcl_abapgit_object_dtdc.clas.abap

```diff
@@ -0,0 +1,456 @@
+
+      clear_field
+        IMPORTING
+          iv_fieldname     TYPE csequence
+        CHANGING
+          cs_dynamic_cache TYPE any,
+
+      fill_metadata_from_db
+        CHANGING
+          cs_dynamic_cache TYPE any
+        RAISING
+          cx_wb_object_operation_error
+          zcx_abapgit_exception,
+
+      get_transport_req_if_needed
+        IMPORTING
+          iv_package                  TYPE devclass
+        RETURNING
+          VALUE(rv_transport_request) TYPE trkorr
+        RAISING
+          zcx_abapgit_exception,
+
+      get_wb_object_operator
+        RETURNING
+          VALUE(ri_wb_object_operator) TYPE REF TO if_wb_object_operator
+        RAISING
+          zcx_abapgit_exception.
+
+    DATA:
+      mr_dynamic_cache      TYPE REF TO data,
+      mv_dynamic_cache_key  TYPE seu_objkey,
+      mi_persistence        TYPE REF TO if_wb_object_persist,
+      mi_wb_object_operator TYPE REF TO if_wb_object_operator.
+
+ENDCLASS.
+
+
+CLASS zcl_abapgit_object_dtdc IMPLEMENTATION.
+
+  METHOD constructor.
+
+    super->constructor(
+        is_item     = is_item
+        iv_language = iv_language ).
+
+    mv_dynamic_cache_key = ms_item-obj_name.
+
+    TRY.
+        CREATE DATA mr_dynamic_cache TYPE ('CL_DTDC_WB_OBJECT_DATA=>TY_DTDC_OBJECT_DATA').
+        CREATE OBJECT mi_persistence TYPE ('CL_DTDC_OBJECT_PERSIST').
+
+      CATCH cx_sy_create_error.
+        zcx_abapgit_exception=>raise( |DTDC not supported by your NW release| ).
+    ENDTRY.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+
+    DATA:
+      li_object_data_model TYPE REF TO if_wb_object_data_model,
+      lx_error             TYPE REF TO cx_wb_object_operation_error.
+
+    TRY.
+        get_wb_object_operator( )->read( IMPORTING eo_object_data = li_object_data_model ).
+
+        rv_user = li_object_data_model->get_changed_by( ).
+
+      CATCH cx_wb_object_operation_error INTO lx_error.
+        zcx_abapgit_exception=>raise(
+            iv_text     = lx_error->get_text( )
+            ix_previous = lx_error ).
+    ENDTRY.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA:
+      lx_error             TYPE REF TO cx_wb_object_operation_error,
+      lv_transport_request TYPE trkorr.
+
+    lv_transport_request = get_transport_req_if_needed( iv_package ).
+
+    TRY.
+        get_wb_object_operator( )->delete( lv_transport_request ).
+
+      CATCH cx_wb_object_operation_error INTO lx_error.
+        zcx_abapgit_exception=>raise(
+            iv_text     = lx_error->get_text( )
+            ix_previous = lx_error ).
+    ENDTRY.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+
+    DATA:
+      li_object_data_model  TYPE REF TO if_wb_object_data_model,
+      li_wb_object_operator TYPE REF TO if_wb_object_operator,
+      lx_error              TYPE REF TO cx_static_check,
+      lv_transport_request  TYPE trkorr.
+
+    FIELD-SYMBOLS:
+      <ls_dynamic_cache> TYPE any,
+      <lv_source>        TYPE data.
```

Comment:
No review, just a question: You seem to differentiate between `any` and `data`. Which behave exactly the same in this context as far as I know. Is there a reason?

Reply:
Nope, I guess the latter one is generated by ADT quick fix and I didn't review its type.